tasks/tests/unit/test_sync_repo_languages_gql.py

```diff
@@ -0,0 +1,121 @@
+    @pytest.mark.asyncio
+    async def test_get_repo_languages_without_org_or_current_owner(self, dbsession):
+        task = SyncRepoLanguagesGQLTask()
+
+        assert await task.run_async(
+            dbsession, org_username="asdf", current_owner_id=123
+        ) == {"successful": False, "error": "no_owner_in_db"}
+
+    @pytest.mark.asyncio
+    async def test_get_repo_languages_with_torngit_rate_limit_error(
+        self, dbsession, mocker, mock_repo_provider
+    ):
+        current_owner = OwnerFactory.create(service="github")
+        org = OwnerFactory.create(service="github")
+
+        dbsession.add_all([current_owner, org])
+        dbsession.flush()
+
+        mocker.patch(
+            f"tasks.{TaskConfigGroup.sync_repo_languages_gql.value}.get_owner_provider_service",
+            return_value=mock_repo_provider,
+        )
+        mock_repo_provider.get_repos_with_languages_graphql = Mock(
+            side_effect=TorngitRateLimitError("response_data", "message", "reset")
+        )
+
+        task = SyncRepoLanguagesGQLTask()
+
+        assert await task.run_async(
+            dbsession, org_username=org.username, current_owner_id=current_owner.ownerid
+        ) == {"successful": False, "error": "torngit_rate_limit_error"}
+
+    @pytest.mark.asyncio
+    async def test_get_repo_languages_with_torngit_error(
+        self, dbsession, mocker, mock_repo_provider
+    ):
+        current_owner = OwnerFactory.create(service="github")
+        org = OwnerFactory.create(service="github")
+
+        dbsession.add_all([current_owner, org])
+        dbsession.flush()
+
+        mocker.patch(
+            f"tasks.{TaskConfigGroup.sync_repo_languages_gql.value}.get_owner_provider_service",
+            return_value=mock_repo_provider,
+        )
+        mock_repo_provider.get_repos_with_languages_graphql = Mock(
+            side_effect=TorngitError()
+        )
+
+        task = SyncRepoLanguagesGQLTask()
+
+        assert await task.run_async(
+            dbsession, org_username=org.username, current_owner_id=current_owner.ownerid
+        ) == {"successful": False, "error": "torngit_error"}
+
+    @pytest.mark.asyncio
+    async def test_get_repo_languages_expected_repsonse(
+        self, dbsession, mocker, mock_repo_provider
+    ):
+        current_owner = OwnerFactory.create(service="github")
+        org = OwnerFactory.create(service="github")
+
+        repo_one_name = "test-one"
+        repo_two_name = "test-two"
+        repo_three_name = "test-three"
+
+        repo_one = RepositoryFactory.create(name=repo_one_name, owner=org)
+        repo_two = RepositoryFactory.create(
+            name=repo_two_name, languages_last_updated=None, owner=org
+        )
+        repo_three = RepositoryFactory.create(name=repo_three_name, owner=org)
+
+        dbsession.add_all([current_owner, org, repo_one, repo_two, repo_three])
+        dbsession.flush()
+
+        MOCKED_NOW = datetime(2024, 7, 3, 6, 8, 12)
```

Comment:
Yeaah I like freeze_gun and tried it here but I was having to assert the DB record was a FakeDateTime, `AssertionError: assert FakeDatetime(2023, 6, 13, 10, 1, 1, 123, tzinfo=datetime.timezone.utc) == '2024-07-03 06:08:12+00:00'` rather than a string like I normally do. It could be an async thing? Not sure but that's why I opted to mock get_utc_now's return value instead, lmk if this feels too odd

Reply:
I see... I think what you have looks perfectly fine.
Given the extra work on the asserting it doesn't pay off to use freeze_gun (since it won't reduce the wordiness of the code anyway)